_code-samples/verify-credential/py/verify_credential.py

```diff
@@ -0,0 +1,168 @@
+        subject - Address of the credential holder/subject, in base58
+        credential_type - Credential type to check for as a string,
+                          which will be encoded as UTF-8 (1-64 bytes long).
+        credential_type_hex - Credential type (binary) as hexadecimal.
+        verbose - If true, print details to stdout during lookup.
+    You must provide either credential_type or credential_type_hex.
+
+    Returns True if the account holds the specified, valid credential.
+    Returns False if the credential is missing, expired, or not accepted.
+    """
+    # Handle function inputs --------------------------------------------------
+    if not (credential_type or credential_type_hex):
+        raise ValueError("Provide a non-empty credential_type or " +
+                         "credential_type_hex")
+    if credential_type and credential_type_hex:
+        raise ValueError("Provide either credential_type or " +
+                         "credential_type_hex, but not both")
+    
+    # Encode credential_type as uppercase hex, if needed
+    if credential_type:
+        credential_type_hex = hexlify(credential_type.encode("utf-8")
+                                      ).decode("ascii")
+        logger.info("Encoded credential_type as hex: "+credential_type_hex.upper())
+    credential_type_hex = credential_type_hex.upper()
+
+    if len(credential_type_hex) % 2 or \
+            not match(r"[0-9A-F]{2,128}", credential_type_hex):
+        # Hexadecimal is always 2 chars per byte, so an odd length is invalid.
+        raise ValueError("credential_type_hex must be 1-64 bytes as hexadecimal.")
+
+    # Perform XRPL lookup of Credential ledger entry --------------------------
+    ledger_entry_request = LedgerEntry(
+        credential={
+            "subject": subject,
+            "issuer": issuer,
+            "credential_type": credential_type_hex
+        },
+        ledger_index="validated"
+    )
+    logger.info("Looking up credential...")
+    logger.info(ledger_entry_request.to_dict())
+    xrpl_response = client.request(ledger_entry_request)
+
+    if xrpl_response.status != "success":
+        if xrpl_response.result["error"] == "entryNotFound":
+            logger.info("Credential was not found")
+            return False
+        # Other errors, for example invalidly-specified addresses.
+        raise XRPLLookupError(xrpl_response)
+
+    credential = xrpl_response.result["node"]
+    logger.info("Found credential:")
+    logger.info(credential)
+
+    # Confirm that the credential has been accepted ---------------------------
+    lsfAccepted = 0x00010000
+    if not credential["Flags"] & lsfAccepted:
+        logger.info("Credential is not accepted.")
+        return False
+    
+    # Confirm that the credential is not expired ------------------------------
+    if credential.get("Expiration"):
+        ## TODO: test this code with expired and non-expired example credentials
```

Comment:
Do we still need this TODO? Seems to work.

Reply:
It does work, as I tested also. Just forgot to remove this. :+1: